common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisExteriorManager.java

```diff
@@ -79,49 +79,26 @@ public TardisExteriorManager(TardisLevelOperator operator) {
         this.operator = operator;
     }
 
-    public void setLastKnownLocation(TardisNavLocation lastKnownLocation) {
-        this.lastKnownLocation = lastKnownLocation;
-    }
-
-    public TardisNavLocation getLastKnownLocation() {
-
-        if(lastKnownLocation == null){
-            return TardisNavLocation.ORIGIN;
-        }
-
-        return this.lastKnownLocation;
-    }
-
-
-    public ServerLevel getLevel() {
-        return this.getLastKnownLocation().getLevel();
-    }
-
-
     @Override
     public void tick() {
 
     }
     @Override
     public CompoundTag saveData(CompoundTag tag) {
 
-        if (this.lastKnownLocation != null) {
-            NbtConstants.putTardisNavLocation(tag, "lk_ext", this.lastKnownLocation);
-        }
         tag.putBoolean(NbtConstants.LOCKED, locked);
 
         return tag;
     }
     @Override
     public void loadData(CompoundTag tag) {
-        this.lastKnownLocation = NbtConstants.getTardisNavLocation(tag, "lk_ext", operator);
         locked = tag.getBoolean(NbtConstants.LOCKED);
     }
 
     public void playSoundAtShell(SoundEvent event, SoundSource source, float volume, float pitch) {
-        if (lastKnownLocation != null) {
-            ServerLevel lastKnownLocationLevel = lastKnownLocation.getLevel();
-            lastKnownLocationLevel.playSound(null, lastKnownLocation.getPosition(), event, source, volume, pitch);
+        if (this.operator.getPilotingManager().getCurrentLocation() != null) {
```

Comment:
getPilotingManager used excessively as getter, should be a local variable
getCurrentLocation used excessively as getter, should be a local variable